src/components/MyAccount/index.tsx

```diff
@@ -197,7 +198,9 @@ const MyAccountContent = ({
   };
 
   const saveProfileChanges = async () => {
-    const updateUserResponse = await updateUserProfile(record);
+    // Save the currently-selected locale, even if it differs from the locale in the profile data we
+    // fetched from the server.
+    const updateUserResponse = await updateUserProfile({ ...record, locale });
```

Comment:
My solution to what we were talking about in Slack. This seemed like the simplest approach. Previously I'd been trying to figure out what sequence of hooks to use to get `record` to always include the correct locale setting, but then I realized the only thing that matters is what locale we submit to the server.

Reply:
This makes sense and totally works. This was my first hunch/suggestion last night on slack.